projects/2022_03_02_Mattermost_Push_Proxy_And_Moblie_Apps.md

```diff
@@ -0,0 +1,169 @@
+
+We would like to implement the push notification system that will enable smartphone users to to receive these notifications, improving the user experience of the PKT chat ecosystem.
+
+## Team and Past Work
+
+The team includes [Christofer Koutelieris](mailto:soricine@gmail.com) and Adonis Gaitatzis. Christopher and Adonis have both worked on PKT mobile software for iOS before an as-yet-unreleased software, and Adonis has worked directly on the PKT blockchain as well as the VPN Utility project, the Telegram trading bot, and the PKT price ticker.
+
+## Project Deliverables
+
+When the project is complete, we will deliver to the pkt.cash operators:
+
+-   A working push-notification server configuration
+-   A PKT-branded Android APK and source code
+-   A PKT-branded iOS IPA and source code
+
+
+The project includes sensitive information including private keys, which should not be revealed to the public. Therefore our deliverables must be handed over to a verified pkt.cash operator rather than open source to the public.
+
+## Success criteria
+
+At the end of this project, we will have:
+
+-   Push notification service running that enables PKT chat users to receive message alerts on their smartphones when they are not using the app.
+
+-   PKT-branded Android and iOS Mattermost apps, which interact with the PKT chat server and the push proxy.
+
+## Payments
+
+* Hourly Rate:
+  * The team's hourly rate is estimated at 2,666.66 PKT/hour based on a PKT price of 0.03 USD / PKT at the time of writing.
+
+
+## Milestones
+
+1.  Push Notification Software
+2.  Push Notification service running online
+3.  Customize and build Android app
+4.  Customize and build iOS app
+
+### Milestone 0 (Kickoff)
+
+We have found the Mattermost instructions for deploying a push notification server, including testing the mobile app on an Android smartphone.
+
+After the project is accepted, the Network Steward will pay XXX PKT
+
+Figure out how time this took, convert that number to PKT using the price at [http://price.pkt.cash](http://price.pkt.cash), then multiply by 120% to get how much PKT we should get paid at this step, then delete this italicized text.
+
+### Milestone 1
+
+#### Push notification software running on a server
+
+At this step we will have the Mattermost push notification software running on a server, but will not be yet connected to a public website.
+
+Estimated man-hours: 15
+
+#### Payout
+
+After a report is delivered on this milestone and it is approved by the NS, the NS will send 40,000 PKT.
+
+
+### Milestone 2
+
+#### Push notification software running on a server
+
+At this step we will connect a web server, likely NGINX to the Mattermost push proxy software, so that the push proxy can operate on the public Internet over secure HTTPS.
+
+Estimated man-hours: 15
+
+#### Payout
+
+After a report is delivered on this milestone and it is approved by the NS, the NS will send 40,000 PKT.
```

Comment:
This is a personal comment and not the opinion of the NS.

I recommend that you specify the cost of each milestone in stable national currency, and report the trading price of PKT *when you deliver the milestone*. This way you will be more or less guaranteed to get paid fairly, and not over-charge in case of price instability.

You will still need establish a maximum cost in PKT of the entire project for NS accounting purposes, but this can be done by adding up your costs at a conservative PKT price.

You may consider looking at this template: https://cryptpad.fr/code/#/2/code/view/lHORne-EdOqRVL2kBQ7SCPpt8U9SoRF5utOCFjnaGVs/ for inspiration about how to structure a project in stable currency and then map it over to PKT.

Reply:
Thankxx for the feedback, I just made the changes .